snakes_cafe.py

```diff
@@ -72,71 +77,133 @@ def ordering():
+    order_string2 = 'Please enter item name and quantity separated by a comma.'
+    order_string3 = 'Enter quit any time to exit.'
+    order_string4 = 'Enter "remove" if you want to remove an item.'
+    print('{} {} {} {}'.format(order_string, order_string2,
+                               order_string3, order_string4))
     while condition:
-            order = input('>' + '\t')
-            order = order.title()
-            if order.split(' ')[0] == 'Quit':
-                condition = False
-                break
-            elif order.split(' ')[0] == 'Order':
-                print_receipt()
-            elif order.split(' ')[0] == 'Remove':
-                for value in menus.values():
-                    for tuple_item in value.keys():
-                        if order.split(' ')[1] == tuple_item[0]:
-                            value[tuple_item] -= 1
-                print_receipt()
-            elif order.split(' ')[0] == 'Menu':
-                menu_items()
-            elif order.split(' ')[0] in menus.keys():
-                for key, value in menus.items():
-                    if order.split(' ')[0] == key:
-                        for key in value.keys():
-                            print(key[0])
+        user_input = input('>' + '\t')
+        user_input = user_input.title()
+        if ',' in user_input:
+            user_input_item, quantity = user_input.split(', ', 1)
+            # check if qty is not negative or string
+            if quantity.isdigit() and int(quantity) > 0:
+                # check stock quantity if valid
+                user_input_qty = int(quantity)
+                if not quantity_check(user_input_qty):
+                    continue
             else:
-                for value in menus.values():
-                    for tuple_item in value.keys():
-                        if order == tuple_item[0]:
-                            value[tuple_item] += 1
-                            print('** ' + str(value[tuple_item]) +
-                                  ' order of ' + tuple_item[0] +
-                                  ' have been added to your meal **')
-                            sub_total()
-                            print('{} {}'.format('Your current total is now: ',
-                                  '$' + str(sub_total())))
+                print('Your entered quantity is not valid.')
+                continue
+        else:
+            user_input_item = user_input
+            user_input_qty = 1
+        if user_input_item == 'Quit':
+            condition = False
+            break
+        elif user_input_item == 'Order':
+            print_receipt()
+        elif user_input_item == 'Remove':
+            print('Enter name of item that you want to remove')
+            which_item_remove = input('>' + '\t')
+            which_item_remove = which_item_remove.title()
+            for key in cart.keys():
+                if which_item_remove in key:
+                    del cart[key]
```

Comment:
You should not be removing the entire order of an item if the user chooses to remove an item. Just one order of it.

Reply:
Also, this is causing an error because you are removing a key as you iterate over the dictionary.